homeassistant/components/wled/light.py

```diff
@@ -73,6 +74,16 @@ async def async_setup_entry(
         "async_effect",
     )
 
+    platform.async_register_entity_service(
+        SERVICE_PRESET,
+        {
+            vol.Required(ATTR_PRESET): vol.All(
+                vol.Coerce(int), vol.Range(min=1, max=16)
```

Comment:
Is this present range fixed and won't change? Or should this also be handled through constants or something that comes back from the API based on the individual configuration of the lights (ie. where a user has only configured 4 presets the accepted range is min=1, max=4).

Reply:
I don't foresee that number expanding due to limitations in the capacity of the ESP8266.

The API doesn't provide the number of slots that have a saved preset in them, either. And in terms of a range, you can save presets to any slot, they don't have to be sequential. Finally, if you try to use a preset where there is nothing defined, nothing happens. So I would say that this is probably the best we can do as a basic safeguard.

Reply:
Well, actually it is -1 to 65535. See WLED API specifications.

Reply:
So in that case @frenck I assume min/max should be updated to -1/65535 correct?
```suggestion
                vol.Coerce(int), vol.Range(min=-1, max=65535)
```

Reply:
From a practical standpoint, I think 1-16 makes the most sense still, and could be expanded if a future version of WLED adds more slots beyond 16. `-1` or `0` will never do anything, `-1` will only be used as a return value if no preset is currently applied.

Reply:
I'd also point out that the `psave` key in the WLED JSON API is documented as only accepting a number between 1-16 for saving a new preset.

Reply:
This is not the psave parameter, this is the ps parameter which doesn't match the specification in this PR. Please adjust, or, if you like it differently; open up a PR with WLED itself to make it 1-16.

Reply:
@frenck Totally good with me - I didn't realize WLED had plans to expand preset slots. I've committed @djtimca's suggestion to make the range match the API then. Since we can't really validate if a preset is currently saved in the existing 16 slots anyways, there's really no additional harm done there from a data validation standpoint.